Description: Use ASM 3.x debian package instead of CGLIB-nodep:
 - change imports to org.objectweb.*
 - small fix to API usage
 - disable build of org.springframework.asm module
Author: Damien Raude-Morvan <drazzib@debian.org>
Last-Update: 2011-08-20
Forwarded: no
--- a/projects/build-spring-framework/build.xml
+++ b/projects/build-spring-framework/build.xml
@@ -25,7 +25,6 @@
 
 	<path id="bundles">
 		<pathelement location="../org.springframework.spring-parent"/>
-		<pathelement location="../org.springframework.asm"/>
 		<path refid="unit.test.bundles" />
 		<pathelement location="../org.springframework.integration-tests"/>
 		<pathelement location="../org.springframework.spring-library"/>
--- a/projects/org.springframework.core/src/main/java/org/springframework/core/LocalVariableTableParameterNameDiscoverer.java
+++ b/projects/org.springframework.core/src/main/java/org/springframework/core/LocalVariableTableParameterNameDiscoverer.java
@@ -28,12 +28,12 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
-import org.springframework.asm.ClassReader;
-import org.springframework.asm.Label;
-import org.springframework.asm.MethodVisitor;
-import org.springframework.asm.Opcodes;
-import org.springframework.asm.Type;
-import org.springframework.asm.commons.EmptyVisitor;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.commons.EmptyVisitor;
 import org.springframework.util.ClassUtils;
 
 /**
@@ -111,7 +111,7 @@
 		try {
 			ClassReader classReader = new ClassReader(is);
 			Map<Member, String[]> map = new ConcurrentHashMap<Member, String[]>();
-			classReader.accept(new ParameterNameDiscoveringVisitor(clazz, map), false);
+			classReader.accept(new ParameterNameDiscoveringVisitor(clazz, map), 0);
 			return map;
 		}
 		catch (IOException ex) {
--- a/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/ClassMetadataReadingVisitor.java
+++ b/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/ClassMetadataReadingVisitor.java
@@ -16,13 +16,13 @@
 
 package org.springframework.core.type.classreading;
 
-import org.springframework.asm.AnnotationVisitor;
-import org.springframework.asm.Attribute;
-import org.springframework.asm.ClassVisitor;
-import org.springframework.asm.FieldVisitor;
-import org.springframework.asm.MethodVisitor;
-import org.springframework.asm.Opcodes;
-import org.springframework.asm.commons.EmptyVisitor;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.Attribute;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.commons.EmptyVisitor;
 import org.springframework.core.type.ClassMetadata;
 import org.springframework.util.ClassUtils;
 
--- a/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationAttributesReadingVisitor.java
+++ b/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationAttributesReadingVisitor.java
@@ -25,9 +25,9 @@
 import java.util.Map;
 import java.util.Set;
 
-import org.springframework.asm.AnnotationVisitor;
-import org.springframework.asm.Type;
-import org.springframework.asm.commons.EmptyVisitor;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.commons.EmptyVisitor;
 import org.springframework.core.annotation.AnnotationUtils;
 import org.springframework.util.ObjectUtils;
 import org.springframework.util.ReflectionUtils;
--- a/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationMetadataReadingVisitor.java
+++ b/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationMetadataReadingVisitor.java
@@ -23,9 +23,9 @@
 import java.util.Map;
 import java.util.Set;
 
-import org.springframework.asm.AnnotationVisitor;
-import org.springframework.asm.MethodVisitor;
-import org.springframework.asm.Type;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Type;
 import org.springframework.core.type.AnnotationMetadata;
 import org.springframework.core.type.MethodMetadata;
 import org.springframework.util.CollectionUtils;
--- a/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MethodMetadataReadingVisitor.java
+++ b/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MethodMetadataReadingVisitor.java
@@ -19,11 +19,11 @@
 import java.util.LinkedHashMap;
 import java.util.Map;
 
-import org.springframework.asm.AnnotationVisitor;
-import org.springframework.asm.MethodAdapter;
-import org.springframework.asm.Opcodes;
-import org.springframework.asm.Type;
-import org.springframework.asm.commons.EmptyVisitor;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.MethodAdapter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.commons.EmptyVisitor;
 import org.springframework.core.type.MethodMetadata;
 import org.springframework.util.MultiValueMap;
 
@@ -95,4 +95,4 @@
 	public String getDeclaringClassName() {
 		return this.declaringClassName;
 	}
-}
\ No newline at end of file
+}
--- a/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReader.java
+++ b/projects/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReader.java
@@ -19,7 +19,7 @@
 import java.io.IOException;
 import java.io.InputStream;
 
-import org.springframework.asm.ClassReader;
+import org.objectweb.asm.ClassReader;
 import org.springframework.core.io.Resource;
 import org.springframework.core.type.AnnotationMetadata;
 import org.springframework.core.type.ClassMetadata;
@@ -51,7 +51,7 @@
 		}
 
 		AnnotationMetadataReadingVisitor visitor = new AnnotationMetadataReadingVisitor(classLoader);
-		classReader.accept(visitor, true);
+		classReader.accept(visitor, ClassReader.SKIP_DEBUG);
 		
 		this.annotationMetadata = visitor;
 		// (since AnnotationMetadataReader extends ClassMetadataReadingVisitor)
@@ -70,4 +70,4 @@
 	public AnnotationMetadata getAnnotationMetadata() {
 		return this.annotationMetadata;
 	}
-}
\ No newline at end of file
+}
--- a/projects/org.springframework.context/src/main/java/org/springframework/scripting/support/ScriptFactoryPostProcessor.java
+++ b/projects/org.springframework.context/src/main/java/org/springframework/scripting/support/ScriptFactoryPostProcessor.java
@@ -20,7 +20,7 @@
 import java.util.Iterator;
 import java.util.Map;
 
-import net.sf.cglib.asm.Type;
+import org.objectweb.asm.Type;
 import net.sf.cglib.core.Signature;
 import net.sf.cglib.proxy.InterfaceMaker;
 import org.apache.commons.logging.Log;
--- a/projects/org.springframework.aop/ivy.xml
+++ b/projects/org.springframework.aop/ivy.xml
@@ -38,7 +38,7 @@
 		<dependency org="org.apache.log4j" name="com.springsource.org.apache.log4j" rev="1.2.15" conf="test->runtime"/>
 		<dependency org="org.easymock" name="com.springsource.org.easymock" rev="2.5.1" conf="test->compile"/>
 		<dependency org="org.junit" name="com.springsource.org.junit" rev="${junit.version}" conf="test->runtime"/>
-		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="test->runtime"/>
+		<!--<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="test->runtime"/>-->
 	</dependencies>
 
 </ivy-module>
--- a/projects/org.springframework.beans/ivy.xml
+++ b/projects/org.springframework.beans/ivy.xml
@@ -32,7 +32,7 @@
 		<dependency org="org.apache.log4j" name="com.springsource.org.apache.log4j" rev="1.2.15" conf="test->runtime"/>
 		<dependency org="org.junit" name="com.springsource.org.junit" rev="${junit.version}" conf="test->runtime"/>
 		<dependency org="org.easymock" name="com.springsource.org.easymock" rev="2.5.1" conf="test->compile"/>
-		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="compile->compile"/>
+		<!--<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="compile->compile"/>-->
 	</dependencies>
 
 </ivy-module>
--- a/projects/org.springframework.context/ivy.xml
+++ b/projects/org.springframework.context/ivy.xml
@@ -51,7 +51,7 @@
 		<dependency org="org.hibernate" name="com.springsource.org.hibernate.validator" rev="4.1.0.GA" conf="optional->compile"/>
 		<dependency org="org.joda" name="com.springsource.org.joda.time" rev="1.6.0" conf="optional->compile"/>
 		<dependency org="org.jruby" name="com.springsource.org.jruby" rev="1.4.0" conf="optional, jruby->compile"/>
-		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="compile->compile"/>
+<!--		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="compile->compile"/>-->
 		<dependency org="org.springframework" name="org.springframework.aop" rev="latest.integration" conf="compile->compile"/>
 		<dependency org="org.springframework" name="org.springframework.beans" rev="latest.integration" conf="compile->compile"/>
 		<dependency org="org.springframework" name="org.springframework.core" rev="latest.integration" conf="compile->compile"/>
--- a/projects/org.springframework.core/ivy.xml
+++ b/projects/org.springframework.core/ivy.xml
@@ -28,7 +28,7 @@
 		<dependency org="org.apache.commons" name="com.springsource.org.apache.commons.logging" rev="1.1.1" conf="compile, commons-logging->compile"/>
 		<dependency org="org.apache.log4j" name="com.springsource.org.apache.log4j" rev="1.2.15" conf="optional, log4j->compile"/>
 		<dependency org="org.aspectj" name="com.springsource.org.aspectj.weaver" rev="${aspectj.version}" conf="optional, aspectj->compile"/>
-		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="optional->compile"/>
+		<!--<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="optional->compile"/>-->
 		<!-- test dependencies -->
 		<dependency org="javax.servlet" name="com.springsource.javax.servlet" rev="2.5.0" conf="test->compile"/>
 		<dependency org="org.junit" name="com.springsource.org.junit" rev="${junit.version}" conf="test->runtime"/>
--- a/projects/org.springframework.integration-tests/ivy.xml
+++ b/projects/org.springframework.integration-tests/ivy.xml
@@ -75,7 +75,7 @@
 		<dependency org="org.hibernate" name="com.springsource.org.hibernate.ejb" rev="3.4.0.GA" conf="test->compile"/>
 		<dependency org="org.hsqldb" name="com.springsource.org.hsqldb" rev="1.8.0.9" conf="test->compile"/>
 		<dependency org="org.jruby" name="com.springsource.org.jruby" rev="1.4.0" conf="optional, runtime->compile"/>
-		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="test->runtime"/>
+<!--		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="test->runtime"/>-->
 		<dependency org="org.springframework" name="org.springframework.aop" rev="latest.integration" conf="test->compile"/>
 		<dependency org="org.springframework" name="org.springframework.beans" rev="latest.integration" conf="test->compile"/>
 		<dependency org="org.springframework" name="org.springframework.context" rev="latest.integration" conf="test->compile"/>
--- a/projects/org.springframework.spring-library/ivy.xml
+++ b/projects/org.springframework.spring-library/ivy.xml
@@ -21,7 +21,7 @@
 
 	<dependencies>
 		<dependency org="org.springframework" name="org.springframework.aop" rev="latest.integration" conf="compile->compile"/>
-		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="compile->compile"/>
+<!--		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="compile->compile"/>-->
 		<dependency org="org.springframework" name="org.springframework.beans" rev="latest.integration" conf="compile->compile"/>
 		<dependency org="org.springframework" name="org.springframework.context" rev="latest.integration" conf="compile->compile"/>
 		<dependency org="org.springframework" name="org.springframework.context.support" rev="latest.integration" conf="compile->compile"/>
--- a/projects/org.springframework.web.portlet/ivy.xml
+++ b/projects/org.springframework.web.portlet/ivy.xml
@@ -35,7 +35,7 @@
 		<dependency org="org.springframework" name="org.springframework.core" rev="latest.integration" conf="compile->compile"/>
 		<dependency org="org.springframework" name="org.springframework.web" rev="latest.integration" conf="compile->compile"/>
 		<dependency org="org.springframework" name="org.springframework.web.servlet" rev="latest.integration" conf="compile->compile"/>
-		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="test->runtime"/>
+		<!--<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration" conf="test->runtime"/>-->
 		<dependency org="org.easymock" name="com.springsource.org.easymock" rev="2.5.1" conf="test->compile"/>
 		<dependency org="org.junit" name="com.springsource.org.junit" rev="${junit.version}" conf="test->runtime"/>
 	</dependencies>
--- a/projects/org.springframework.web.servlet/ivy.xml
+++ b/projects/org.springframework.web.servlet/ivy.xml
@@ -84,8 +84,8 @@
 		<dependency org="org.easymock" name="com.springsource.org.easymock" rev="2.5.1" conf="test->compile"/>
 		<dependency org="org.apache.commons" name="com.springsource.org.apache.commons.fileupload" rev="1.2.0"
                     conf="test->compile"/>		
-		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration"
-                    conf="test->compile"/>
+<!--		<dependency org="org.springframework" name="org.springframework.asm" rev="latest.integration"
+                    conf="test->compile"/>-->
 		<dependency org="org.custommonkey.xmlunit" name="com.springsource.org.custommonkey.xmlunit" rev="1.2.0"
                     conf="test->compile"/>
 		<dependency org="org.dom4j" name="com.springsource.org.dom4j" rev="1.6.1" conf="test->compile"/>
